Sources/MatrixClient/API/Room/CreateRoom.swift

```diff
@@ -0,0 +1,178 @@
+//
+//  File.swift
+//
+//
+//  Created by Finn Behrens on 13.03.22.
+//
+
+import AnyCodable
+import Foundation
+
+// TODO: Links
+public struct MatrixCreateRoomRequest: Codable {
+    /// Extra keys, such as `m.federate`, to be added to the content of the
+    /// ``MatrixClient/ event. The server will overwrite the following keys: creator, room_version. Future versions of the specification may allow the server to overwrite other keys.
+    public var creationContent: MatrixRoomCreateEvent.Content?
+
+    /// A list of state events to set in the new room.
+    ///
+    /// This allows the user to override the default state events set in the new room.
+    /// The expected format of the state events are an object with type, state_key and
+    /// content keys set.
+    ///
+    /// Takes precedence over events set by preset, but gets overridden by name and topic keys.
+    public var initialState: [StateEvent]?
+
+    /// A list of user IDs to invite to the room.
+    ///
+    /// This will tell the server to invite everyone in the list to the newly created room.
+    public var invite: [String]?
+
+    /// A list of objects representing third party IDs to invite into the room.
+    public var invite3Pid: [Invite3pid]?
```

Comment:
The 3pid casing doesn’t match between the name and the type. I’d probably favour `3PID` myself, but easy on this one.